File: java/playwright/src/test/java/dev/selenium/BasicAuthTest.java
package dev.selenium;

import com.microsoft.playwright.Browser;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.*;

public class BasicAuthTest extends TestBase {

  @Test
  public void basicAuthContext() {
    Browser.NewContextOptions options =
        new Browser.NewContextOptions().setHttpCredentials("admin", "admin");

    context.close();
    context = browser.newContext(options);
    page = context.newPage();

    page.navigate("https://the-internet.herokuapp.com/basic_auth");

    Assertions.assertEquals(
        "Congratulations! You must have the proper credentials.", page.locator("p").innerText());
  }
}

File: dotnet/Playwright/BasicAuthTest.cs
using Microsoft.Playwright;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Playwright;

[TestClass]
public class BasicAuthTest : TestBase
{
    [TestMethod]
    public async Task BasicAuthContext()
    {
        var options = new BrowserNewContextOptions
        {
            HttpCredentials = new HttpCredentials
            {
                Username = "admin",
                Password = "admin"
            }
        };

        await Context!.CloseAsync();
        Context = await Browser!.NewContextAsync(options);
        Page = await Context.NewPageAsync();

        await Page.GotoAsync("https://the-internet.herokuapp.com/basic_auth");

        var paragraphText = await Page.Locator("p").InnerTextAsync();
        Assert.AreEqual(
            "Congratulations! You must have the proper credentials.",
            paragraphText);
    }
}